### PR TITLE
2443 outline label too long

### LINKED ISSUE #2443: Implementation labels in outline view are too long
## Summary

The label text shown in the outline view is often so long that the name of the element is cut off at the right.

## Expected and Current Behavior

The fixed text part of the label, e.g., _Subprogram Group Impl_, or _Flow Implementation_ should be shortened to allow more space for the name.

Is it possible to use an icon decoration (like the navigator uses for project type, etc.) to indicate that something is an implementation?

## Environment

* **OSATE Version**: 2.8.0
* **Operating System**: all

### PR DESCRIPTION
**Pull request by asazonova**
Add letter I to images for implementation elements in outline view
Change color of text for implementation elements in outline view

fixes #2443

### CODE REVIEW

**Review by lwrage — CHANGES_REQUESTED** [review]
There are many more labels that could be shortened, e.g., _Subcomponent_ or _End to end flow_. Could you add a list of labels to the issue and propose shorter versions, please? Then we can discuss.

File: core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/AnnexAwareDeclarativeLabelProvider.java [hunk 1]
```diff
@@ -51,14 +67,24 @@ public AnnexAwareDeclarativeLabelProvider(ILabelProvider delegate) {
...
+					|| element instanceof ProcessorImplementation || element instanceof ProcessImplementation
+					|| element instanceof ThreadGroupImplementation || element instanceof ThreadImplementation
+					|| element instanceof SubprogramImplementation || element instanceof SubprogramGroupImplementation
+					|| element instanceof AbstractImplementation || element instanceof BusImplementation
+					|| element instanceof MemoryImplementation || element instanceof DeviceImplementation
+					|| element instanceof VirtualBusImplementation || element instanceof VirtualProcessorImplementation
+					|| element instanceof FlowImplementation) {
+				StyledString str = this.convertToStyledString(
+						text.toString().replaceAll("Implementation", "").replaceAll("Impl", "").replaceAll("  ", " "));
+				str.setStyle(0, str.length(),
```
Comment (lwrage) [inline core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/AnnexAwareDeclarativeLabelProvider.java:83]: Avoid color changes. What looks OK in a dark theme may look bad in a light theme and vice versa.

**Comment by asazonova** [comment]
Lutz, it looks like when I synced the branch, changes from file core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/Aadl2LabelProvider.java      got pulled in to this PR. I sort of assumed that this code is already in master branch, but maybe not...

File: core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/Aadl2LabelProvider.java [hunk 2]
```diff
@@ -391,6 +393,30 @@ String text(Feature ele) {
 		return "Feature " + ele.getName();
 	}
 
+	String text(FlowImplementation flowimpl) {
+		String ret;
+
+		ret = "Flow ";
```
Comment (lwrage) [inline core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/Aadl2LabelProvider.java:399]: Remove space character from this string.

File: core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/AnnexAwareDeclarativeLabelProvider.java [hunk 3]
```diff
@@ -51,14 +65,20 @@ public AnnexAwareDeclarativeLabelProvider(ILabelProvider delegate) {
...
 		if (text != null) {
+			if (element instanceof SystemImplementation || element instanceof DataImplementation
+					|| element instanceof ProcessorImplementation || element instanceof ProcessImplementation
+					|| element instanceof ThreadGroupImplementation || element instanceof ThreadImplementation
+					|| element instanceof SubprogramImplementation || element instanceof SubprogramGroupImplementation
+					|| element instanceof AbstractImplementation || element instanceof BusImplementation
+					|| element instanceof MemoryImplementation || element instanceof DeviceImplementation
+					|| element instanceof VirtualBusImplementation || element instanceof VirtualProcessorImplementation
+					|| element instanceof FlowImplementation) {
+
```
Comment (lwrage) [inline core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/AnnexAwareDeclarativeLabelProvider.java:79]: As you list ALL sublasses of ComponentImplementation here, you can replace the condition with if (element instanceof ComponentImplementation)

**Review by lwrage — CHANGES_REQUESTED** [review]
Please try a bigger I in the icons it's currently too small. Can you put the I in the middle of each icon? make it fatter? maybe add a horizontal bar at the top and bottom?

**Comment by asazonova** [comment]
@lwrage , i finally found a tool that allows to create a 16x16 pixel icon. I also tried making an image with different app and then resizing to 16x16 pixels, but that resulted in blurry unrecognizable images. I have created a new image for system implementation via https://www.pixilart.com/ website. It does allow to fill each pixel one by one, so I will easily be able to keep an I letter in the same space/size. I committed the changes, let me know if it looks okay

File: core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/AnnexAwareDeclarativeLabelProvider.java (hunk 3 above)
Comment (asazonova) [reply core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/AnnexAwareDeclarativeLabelProvider.java]: changed to ComponentImplementation

File: core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/Aadl2LabelProvider.java (hunk 2 above)
Comment (asazonova) [reply core/org.osate.xtext.aadl2.ui/src/org/osate/xtext/aadl2/ui/labeling/Aadl2LabelProvider.java]: deleted extra space

**Review by lwrage — APPROVED** [review]

**Pull request merged**